### PR TITLE
feat(cilium): Create bpffs mount on every node as cilium (almost) requires it

### PR DESCRIPTION
**Pull request by skinny**
This PR allows for the creation of a new systemd unit to auto-mount the bpffs (almost) required by the Cilium project (http://docs.cilium.io/en/latest/kubernetes/configuration/?highlight=bpffs#mounting-bpffs-with-systemd). After creating the unit the REBOOT_REQUIRED flag is set to ensure mounting occurs (had some trouble with only enable/restarting the mount).

Reason for this PR is to allow the cilium daemonset to work properly out of the box. More steps are needed to create a 100% experience the first time but this is a step towards that goal.

### CODE REVIEW

**Comment by jackfrancis** [comment]
Hi @skinny, is this related to https://github.com/Azure/aks-engine/issues/112

Can you provide a (scrubbed of creds) working cilium-backed api model configuration that we can use to start testing regularly?

**Comment by skinny** [comment]
Hi @jackfrancis! Currently I'm just testing with the AKS-engine 101 model with only the `networkPlugin` set to `cilium` .  

I leave out the `networkPolicy` for now so I can deploy the latest (currently custom) Cilium yamls on top of that. I'm pretty sure that the current `kubernetesmasteraddons-cilium-daemonset.yaml` will work fine too but I have not tested that yet. Next PR will include a new version of that file so you can also set the 'networkPolicy' to 'cilium'.

**Comment by skinny** [comment]
```{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorVersion": "1.13.2",
      "kubernetesConfig": {
        "serviceCidr": "10.1.0.0/16",
        "dnsServiceIP": "10.1.0.10",
        "clusterSubnet": "10.244.0.0/16",
        "etcdDiskSizeGB": "64",
        "networkPlugin": "cilium",
        "apiServerConfig": {
          "--enable-admission-plugins": "NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,AlwaysPullImages"
        },
        "addons": [{
          "name": "tiller",
          "enabled": true
        }, {
          "name": "ip-masq-agent",
          "enabled": false
        }]
      }
    },
    "masterProfile": {
      "count": 3,
      "vmSize": "Standard_A2_v2",
      "vnetSubnetId": ""
    },
    "agentPoolProfiles": [{
      "name": "default",
      "count": 3,
      "vmSize": "Standard_D2_v2"
    }],
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [{
          "keyData": "..."
        }]
      }
    }
  }
}```

**Comment by jackfrancis** [comment]
@skinny I'm not able to build a functional cluster out of the box against this PR using the following api model:

```
{   "apiVersion": "vlabs",   "properties": {     "orchestratorProfile": {       "orchestratorType": "Kubernetes", "orchestratorRelease": "1.13",       "kubernetesConfig": {         "networkPlugin": "cilium", "apiServerConfig": { "--enable-admission-plugins": "NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,AlwaysPullImages" }, "clusterSubnet": "10.239.0.0/16",         "addons": [           {             "name": "tiller",             "enabled": true }, { "name": "ip-masq-agent", "enabled": false }             ]           }                         },     "masterProfile": {       "count": 3,       "dnsPrefix": "",       "vmSize": "Standard_D2_v2",       "OSDiskSizeGB": 200,       "vnetSubnetId": "/subscriptions/SUB_ID/resourceGroups/RG_NAME/providers/Microsoft.Network/virtualNetworks/VNET_NAME/subnets/SUBNET_NAME",       "firstConsecutiveStaticIP": "10.239.255.239",       "vnetCidr": "10.239.0.0/16"     },     "agentPoolProfiles": [       {         "name": "agent1",         "count": 3,         "vmSize": "Standard_D2_v2",         "OSDiskSizeGB": 200,         "storageProfile": "ManagedDisks",         "diskSizesGB": [           128,           128,           128,           128         ],         "availabilityProfile": "AvailabilitySet",         "vnetSubnetId": "/subscriptions/SUB_ID/resourceGroups/RG_NAME/providers/Microsoft.Network/virtualNetworks/VNET_NAME/subnets/SUBNET_NAME"       },       {         "name": "agent2",         "count": 3,         "vmSize": "Standard_D2_v3",         "OSDiskSizeGB": 200,         "storageProfile": "ManagedDisks",         "diskSizesGB": [           128         ],         "availabilityProfile": "AvailabilitySet",         "vnetSubnetId": "/subscriptions/SUB_ID/resourceGroups/RG_NAME/providers/Microsoft.Network/virtualNetworks/VNET_NAME/subnets/SUBNET_NAME"       }     ],     "linuxProfile": {       "adminUsername": "azureuser",       "ssh": {         "publicKeys": [           {             "keyData": ""           }         ]       }     },     "servicePrincipalProfile": {       "clientId": "",       "secret": ""     }   } }
```

I think the above is functionally equivalent to yours.

tl;dr is below:

```
$ kubectl get nodes
2019/02/07 21:14:15 NAME                    STATUS     ROLES     AGE       VERSION   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
k8s-agent1-27343826-0   NotReady   agent     31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-agent1-27343826-1   NotReady   agent     31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-agent1-27343826-2   NotReady   agent     31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-agent2-27343826-0   NotReady   agent     31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-agent2-27343826-1   NotReady   agent     31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-agent2-27343826-2   NotReady   agent     31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-master-27343826-0   NotReady   master    31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-master-27343826-1   NotReady   master    31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
k8s-master-27343826-2   NotReady   master    31m       v1.13.3   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   docker://3.0.1
```

**Comment by skinny** [comment]
@jackfrancis The 'NotReady' nodes are to be expected as I left out the `networkPlugin` to allow me to deploy my latest cilium DaemonSet  manually. 

Let me try to get you a real working sample tomorrow.

**Comment by skinny** [comment]
replaced by https://github.com/Azure/aks-engine/pull/508
